### DISCUSSION TITLE
Using global graph features for a node level task

### DISCUSSION THREAD

**Discussion by alzaia** [pyg-team/pytorch_geometric#5910] [Q&A]
Hello, I am working on a node prediction use case. I am already using the node attributes `x` and the edge attributes `edge_attr` in my models. However, I would also want to try to incorporate graph info (e.g. different types of graphs could need adjustment in terms of node prediction). I could simply broadcast the graph feature(s) into each node, but I'm trying to find something more elegant. First, are there any models/examples in PyG that already do that combination of global and local features? If not, how could I change my GNN architecture to adapt it to this problem? Thanks!

**Comment by rusty1s** [comment]
I think the idea of broadcasting is pretty common. Alternatively, you can add a virtual node to the graph that is connected to all other nodes, and fill it with global graph information, but this might require a heterogeneous GNN to treat this type of node differently.

**Reply by alzaia** [reply]
Thanks a lot for your great input @rusty1s ! I just discovered the `MetaLayer` which could be the way to solve my needs eventually in a more elegant way.
